crates/core/src/diff/fragment/mod.rs

```diff
@@ -81,17 +81,25 @@ impl Fragment {
             Fragment::Regular { children, statics } => {
                 match statics {
                     Statics::Statics(statics) => {
-                        assert!(statics.len() == children.len() + 1);
                         out.push_str(&statics[0]);
                         for i in 1..statics.len() {
-                            let child = children.get(&(i - 1).to_string()).expect("Failed to get child");
+                            let child = children.get(&(i - 1).to_string()).ok_or(RenderError::ChildNotFoundForStatic((i - 1) as i32))?;
                             let val = child.render(components, cousin_statics.clone(), parent_templates.clone())?;
                             out.push_str(&val);
                             out.push_str(&statics[i]);
                         }
                     }
-                    Statics::TemplateRef(_template_ref) => {
-                        unimplemented!("Template id's in regular fragments have not been implement yet");
+                    Statics::TemplateRef(template_id) => {
+                        let templates = parent_templates.ok_or(RenderError::NoTemplates)?;
+                        let template = templates.get(&(template_id.to_string())).ok_or(RenderError::TemplateNotFound(*template_id))?;
+                        out.push_str(&template[0]);
+                        for i in 1..template.len() {
+                            let child_id = i - 1;
+                            let child = children.get(&child_id.to_string()).ok_or(RenderError::ChildNotFoundForTemplate(child_id as i32))?;
+                            let val = child.render(components, cousin_statics.clone(), Some(templates.clone()))?;
+                            out.push_str(&val);
+                            out.push_str(&template[i]);
+                        }
```

Comment:
Perhaps a comment indicating why we're starting this iterator at 1 instead of 0 would be helpful. It's not immediately clear to me reading this code.